CHANGELOG.md

```diff
@@ -1,20 +1,28 @@
 # Release History
 
-## 1.17.0 (TBD)
+## 1.17.0 (2024-05-21)
 
-### New Features
+### Snowpark Python API Updates
+
+#### New Features
 
 - Added support to add a comment on tables and views using functions listed below:
   - `DataFrameWriter.save_as_table`
   - `DataFrame.create_or_replace_view`
   - `DataFrame.create_or_replace_temp_view`
   - `DataFrame.create_or_replace_dynamic_table`
 
-### Improvements
+#### Improvements
 
 - Improved error message to remind users set `{"infer_schema": True}` when reading csv file without specifying its schema.
 
-### Local Testing Updates
+### Snowpark pandas API updates
```

Comment:
nit: here it is `updates` but in the other heading it is `Updates`. Maybe let's stick with one?